### PR TITLE
Cleanup script for removing all instances and disks in the cloud

### PR DESCRIPTION
**Pull request by stejskalleos**

### CODE REVIEW

**Comment by adamruzicka** [comment]
Just to make sure I'm reading this right, the intent of the script is to remove *everything*, correct?

**Comment by stejskalleos** [comment]
> Just to make sure I'm reading this right, the intent of the script is to remove _everything_, correct?

Yop, sometimes I forgot to remove all the instances (and their disks) I created during the day and then the billing is going up faster than I want to.

Plus it's way faster than the Google's UI.

**Comment by adamruzicka** [comment]
Needs a rebase now

File: scripts/cleanup_cloud.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+client = ForemanGoogle::GoogleComputeAdapter.new auth_json_string: File.read(path)
+
+instances = client.instances(zone)
+
+puts 'Found instances:'
+puts 'Nothing found' if instances.empty?
+
+instances.each do |instance|
+  puts ">>> Deleting #{instance.name} instance"
+  client.delete_instance(zone, instance.name)
```
Comment (adamruzicka) [inline scripts/cleanup_cloud.rb]: It feels like deletion of an instance is an asynchronous operation and disk removal may fail if the instance removal is not fully finished yet.

```
$ bundle exec ruby scripts/cleanup_cloud.rb  /home/aruzicka/Downloads/noted-terra-303613-69ae49292458.json     us-central1-a
Found instances:
>>> Deleting instance-1 instance

Found disks:
>>> Deleting disk-1 disk
Traceback (most recent call last):
        17: from scripts/cleanup_cloud.rb:30:in `<main>'
        16: from scripts/cleanup_cloud.rb:30:in `each'
        15: from scripts/cleanup_cloud.rb:32:in `block in <main>'
        14: from /home/aruzicka/vcs/foreman/foreman-google/app/lib/foreman_google/google_compute_adapter.rb:76:in `delete_disk'
        13: from /home/aruzicka/vcs/foreman/foreman-google/app/lib/foreman_google/google_compute_adapter.rb:115:in `delete'
        12: from /home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/google-cloud-compute-v1-1.3.0/lib/google/cloud/compute/v1/disks/rest/client.rb:469:in `delete'
        11: from /home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/google-cloud-compute-v1-1.3.0/lib/google/cloud/compute/v1/disks/rest/service_stub.rb:202:in `delete'
        10: from /home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/gapic-common-0.9.0/lib/gapic/rest/client_stub.rb:76:in `make_delete_request'
         9: from /home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/gapic-common-0.9.0/lib/gapic/rest/client_stub.rb:129:in `make_http_request'
         8: from /home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/faraday-1.10.0/lib/faraday/connection.rb:200:in `delete'
         7: from /home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/faraday-1.10.0/lib/faraday/connection.rb:516:in `run_request'
         6: from /home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/faraday-1.10.0/lib/faraday/rack_builder.rb:154:in `build_response'
         5: from /home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/gapic-common-0.9.0/lib/gapic/rest/faraday_middleware.rb:38:in `call'
         4: from /home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/faraday-retry-1.0.3/lib/faraday/retry/middleware.rb:140:in `call'
         3: from /home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/faraday-1.10.0/lib/faraday/middleware.rb:18:in `call'
         2: from /home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/faraday-1.10.0/lib/faraday/response.rb:61:in `on_complete'
         1: from /home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/faraday-1.10.0/lib/faraday/middleware.rb:19:in `block in call'
/home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/faraday-1.10.0/lib/faraday/response/raise_error.rb:16:in `on_complete': the server responded with status 400 (Faraday::BadRequestError)
        6: from scripts/cleanup_cloud.rb:30:in `<main>'
        5: from scripts/cleanup_cloud.rb:30:in `each'
        4: from scripts/cleanup_cloud.rb:32:in `block in <main>'
        3: from /home/aruzicka/vcs/foreman/foreman-google/app/lib/foreman_google/google_compute_adapter.rb:76:in `delete_disk'
        2: from /home/aruzicka/vcs/foreman/foreman-google/app/lib/foreman_google/google_compute_adapter.rb:115:in `delete'
        1: from /home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/google-cloud-compute-v1-1.3.0/lib/google/cloud/compute/v1/disks/rest/client.rb:446:in `delete'
/home/aruzicka/.toolbox/f35/.rvm/gems/ruby-2.7.2/gems/google-cloud-compute-v1-1.3.0/lib/google/cloud/compute/v1/disks/rest/client.rb:484:in `rescue in delete': An error has occurred when making a REST request: The disk resource 'projects/noted-terra-303613/zones/us-central1-a/disks/disk-1' is already being used by 'projects/noted-terra-303613/zones/us-central1-a/instances/instance-1' (Google::Cloud::InvalidArgumentError)
        5: from scripts/cleanup_cloud.rb:30:in `<main>'
        4: from scripts/cleanup_cloud.rb:30:in `each'
        3: from scripts/cleanup_cloud.rb:32:in `block in <main>'
        2: from /home/aruzicka/vcs/foreman/foreman-google/app/lib/foreman_google/google_compute_adapter.rb:76:in `delete_disk'
        1: from /home/aruzicka/vcs/foreman/foreman-google/app/lib/foreman_google/google_compute_adapter.rb:114:in `delete'
/home/aruzicka/vcs/foreman/foreman-google/app/lib/foreman_google/google_compute_adapter.rb:117:in `rescue in delete': uninitialized constant ForemanGoogle::GoogleComputeAdapter::Foreman (NameError)
```

+ the name error is something we should address

**Review by adamruzicka — CHANGES_REQUESTED** [review]

File: app/lib/foreman_google/google_compute_adapter.rb [hunk 2]
```diff
@@ -48,6 +48,10 @@ def stop(zone, instance_identity)
       manage_instance(:stop, zone: zone, instance: instance_identity)
     end
 
+    def delete_instance(zone, instance_identity)
```
Comment (stejskalleos) [inline app/lib/foreman_google/google_compute_adapter.rb:51]: `delete_instance` and `disks` are  not used anymore by the cleanup_script but they'll be useful later in follow up PRs

**Review by adamruzicka — COMMENTED** [review]
The asynchronous nature of removal still remains unaddressed. The error is nicer now, but I still have to run the script, wait for a minute or two and then run it again to really remove everything.

```shell
# bundle exec ruby scripts/cleanup_cloud.rb /home/aruzicka/Downloads/noted-terra-303613-69ae49292458.json us-central1-a
Deleting instances:
>>> Deleting instance-1 instance

Deleting disks:
>>> Deleting disk-1 disk
ERROR: An error has occurred when making a REST request: The disk resource 'projects/noted-terra-303613/zones/us-central1-a/disks/disk-1' is already being used by 'projects/noted-terra-303613/zones/us-central1-a/instances/instance-1'
>>> Deleting instance-1 disk
ERROR: An error has occurred when making a REST request: The disk resource 'projects/noted-terra-303613/zones/us-central1-a/disks/instance-1' is already being used by 'projects/noted-terra-303613/zones/us-central1-a/instances/instance-1'

Done.
```

**Pull request closed without merging**